omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/PatientAllergyController2_0Test.java

```diff
@@ -0,0 +1,346 @@
+
+		// attempt to delete all allergies
+		handle(newDeleteRequest(getURI(), new Parameter("reason", "unit test")));
+		
+		allergy = Context.getPatientService().getAllergyByUuid(getUuid());		
+		allergies = Context.getPatientService().getAllergies(allergy.getPatient());
+		Assert.assertTrue(allergy.isVoided());
+		Assert.assertEquals("unit test", allergy.getVoidReason());
+		Assert.assertEquals(Allergies.UNKNOWN, allergies.getAllergyStatus());
+	}
+
+	/*
+	 *  Get Allergy 
+	 */
+	@Test
+	public void shouldGetAllergyByUuid() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+
+		// attempt to get allergy by uuid
+		SimpleObject so = deserialize(handle(newGetRequest(getURI() + "/" + getUuid())));
+
+		Assert.assertEquals(allergy.getComment(), Util.getByPath(so, "comment"));
+		Assert.assertEquals(allergy.getSeverity().getUuid(), Util.getByPath(so, "severity/uuid"));
+
+		if (allergy.getAllergen().isCoded()) {
+			Assert.assertEquals(allergy.getAllergen().getCodedAllergen().getUuid(), Util.getByPath(so, "allergen/codedAllergen/uuid"));
+		} else {
+			Assert.assertEquals(allergy.getAllergen().getCodedAllergen().getUuid(), Util.getByPath(so, "allergen/codedAllergen/uuid"));
+			Assert.assertEquals(allergy.getAllergen(), Util.getByPath(so, "allergen/nonCodedAllergen"));
+		}
+		Assert.assertEquals(allergy.getAllergen().getAllergenType().toString(), Util.getByPath(so, "allergen/allergenType"));
+	}
+
+	/**
+	 *  Save Allergy with Coded Allergen
+	 */
+	@Test
+	public void shouldSaveAllergyWithCodedAllergen() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+		Allergies allergies = Context.getPatientService().getAllergies(patient);
+		Assert.assertEquals(allergies.size(), 4);
+		
+		// save allergy with coded allergen
+		String json = "{" +
+				        " \"comment\" : \"allergy comment\"," +
+				        " \"severity\" : { \"uuid\" : \"35d3346a-6769-4d52-823f-b4b234bac3e3\"}," +
+				        " \"allergen\" : " + 
+				          " { \"allergenType\" : \"DRUG\", " +
+				          " \"codedAllergen\" : { \"uuid\" : \"35d3346a-6769-4d52-823f-b4b234bac3e3\"} " +
+				          " } " +
+		              "}";
+		
+		// save allergy with reactions to get uuid for later saving of reactions
+		SimpleObject so = deserialize(handle(newPostRequest(getURI(), json)));
+		
+		// update allergy with reactions
+		json = "{ \"reactions\" : " +
+   			       "[" +
+			         "{" +
+					   " \"allergy\" : { \"uuid\" : \"" + PropertyUtils.getProperty(so, "uuid") + "\"}," +
+			           " \"reaction\" : { \"uuid\" : \"35d3346a-6769-4d52-823f-b4b234bac3e3\" }" +
+			         "}," +
+			         "{" +
+					   " \"allergy\" : { \"uuid\" : \"" + PropertyUtils.getProperty(so, "uuid") + "\"}," +
+			           " \"reaction\" : { \"uuid\" : \"5622AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA\" }," +

```

Comment:
Can you make this test method simpler by leaving it was it was. To assert that a coded allergen is saved properly.
Then create a separate test method for reactions. Where you assert the reactions using the return value for: allergies = Context.getPatientService().getAllergies(patient)